test/Data/String/ConversionsSpec.hs

```diff
@@ -10,11 +11,184 @@ import           Data.String.Conversions
 spec :: Spec
 spec = do
   describe "cs" $ do
-    it "allows to convert from String to strict Text" $ do
-      property $ \ s ->
-        show (s :: String) `shouldBe` show (cs s :: ST)
-
-    context "when converting from strict ByteString to strict Text" $ do
-      it "never crashes" $ do
-        property $ \ s ->
-          deepseq (cs (s :: SBS) :: ST) (return () :: IO ())
+    context "losslessly" $ do
+      it "converts String -> String -> String" $ do
```

Comment:
Could we add more sub-`context`s (or `describe`s) under `"losslessly"` for the groups of tests that roundtrip the same type? E.g.
``` haskell
describe "roundtripping String" $ do
  ...
describe "roundtripping StrictText" $ do
  ...
...
```

I think that would make it easier to understand which combinations are tested and which aren't.